source/client.js

```diff
@@ -410,6 +410,22 @@ export class ImmersClient extends window.EventTarget {
     return this.activities.note(DOMPurify.sanitize(content), to, privacy)
   }
 
+  /**
+   * Delete  user's avatar from their Avatar Collection.
+   * @param  {(object|string)} sourceActivity - IRI of activity or Activity in the Outbox
+   * @returns {Promise<string>} IRI of the remove activity
+   */
+  async deleteMessage (sourceActivity) {
+    switch (sourceActivity.type.toLowerCase()) {
+      case 'arrive':
+        return this.activities.undo(sourceActivity)
+      case 'leave':
+        return this.activities.undo(sourceActivity)
+      case 'create':
+        return this.activities.delete(sourceActivity)
```

Comment:
I wonder if we should be more flexible and just fall back to undo as a `default` for any types that we don't have special handling for (I suppose just Create at this time)